### ISSUE TITLE
Top level sections are not accessible and don't have content when accessed directly

### ISSUE THREAD

**Issue by molant** [webhintio/webhint.io#520]
The content of `development flow integration` is:

![image](https://user-images.githubusercontent.com/606594/45786304-b9644180-bc24-11e8-89bb-9c00a3d7cdf0.png)

But the [`index.md` has content](https://github.com/webhintio/hint/blob/master/packages/hint/docs/user-guide/development-flow-integration/index.md).

Also because of how navigation works the user needs to manually type the url, there's no way to access it otherwise.

As a worse example, `Telemetry` is a category that doesn't have any content.

We should:

* [ ] Figure out a way to make the menu work that it's also accessible
* [ ] Fix the issue with `index.md` not displaying content for those pages

**Comment by molant** [comment]
Changing the name of the file from `index.md` to `summary.md` outputs the following:

![image](https://user-images.githubusercontent.com/606594/47330657-e2b22c00-d62d-11e8-805b-c01ccf697433.png)

Few issues/concerns:

* It should always be the first item
* Are we ok with the current names? Do we want to update them to summary?
